beta/scripts/og-generator/index.js

```diff
@@ -0,0 +1,43 @@
+const fm = require('gray-matter');
+const {getCss, getHTML} = require('./util');
+const globby = require('globby');
+const fs = require('fs-extra');
+const slugs = require('github-slugger');
+const core = require('puppeteer-core');
+const chrome = require('chrome-aws-lambda');
+let _page;
+
+const getPage = async (isDev) => {
+  if (_page) {
+    return _page;
+  }
+  const options = {
+    args: chrome.args,
+    executablePath: await chrome.executablePath,
+    headless: chrome.headless,
+  };
+  const browser = await core.launch(options);
+  _page = await browser.newPage();
+  return {page: _page, browser};
+};
+
+(async () => {
+  const markdownPages = await globby('src/content/**/*.{md,mdx}');
+  //   launch browser
+  const {page, browser} = await getPage(false);
+  await page.setViewport({width: 800, height: 400});
+  for (let filepath of markdownPages) {
+    const rawStr = await fs.readFile(filepath, 'utf8');
+    // Extract frontmatter
+    const {
+      data: {title, description},
```

Comment:
We need to extract description from MDX. You can do this by parsing MDX first and then looking for `Intro` element there. See `prepareMDX` for similar existing code.

Reply:
Hm I guess if we do include descriptions, the images would need to be regenerated when you edit one. That's a bit annoying to keep track of. 🤔 Maybe we can just set up a task to do this once a day or something.

Reply:
how do we expect the images when descriptions are too long? they wont fit inside the image?

Reply:
Hmm. Yeah I don't know. 😄 

How does the Vercel thing deal with it?

Reply:
It breaks 😅  https://og-image.vercel.app/**Hello**%20World**Hello**%20World**Hello**%20World**Hello**%20World**Hello**%20World**Hello**%20World**Hello**%20World**Hello**%20World**Hello**%20World**Hello**%20World**Hello**%20World**Hello**%20World**Hello**%20World**Hello**%20World**Hello**%20World**Hello**%20World**Hello**%20World**Hello**%20World.png?theme=light&md=1&fontSize=100px&images=https%3A%2F%2Fassets.vercel.com%2Fimage%2Fupload%2Ffront%2Fassets%2Fdesign%2Fvercel-triangle-black.svg

Reply:
>how about we add min-height for headings and descriptions display an ellipsis (...)? would that be good?

seems fine to try

Reply:
These intro components have markdown content inside it. Like https://github.com/reactjs/reactjs.org/blob/b6d597dba30fcec25944395883714eba27e997bd/beta/src/content/apis/react-dom/hydrate.md?plain=1#L10-L18

How are we going trying to address these inside descriptions?

Reply:
Can we just render them to HTML and display as HTML? I.e. using the regular MDX pipeline.

Reply:
This is how it looks when we have text + code snippet inside description....
<img width="646" alt="Screenshot 2022-09-09 at 1 05 41 PM" src="https://user-images.githubusercontent.com/32865581/189297170-e3723ee1-40b0-42e5-9304-41356b1b64c2.png">

Title with ellipsis? 
<img width="620" alt="Screenshot 2022-09-09 at 1 06 16 PM" src="https://user-images.githubusercontent.com/32865581/189297269-f6acf15e-95ce-4bbe-be6e-839c50c6a739.png">

Some blogs don't have an intro section. so may be at that point of time we might want to alter the layout of og-image? since it looks too empty?

Reply:
PS: these are edge cases others look good